src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AddMissingInterpolationTokenTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines.CSharpAddMissingInterpolationToken,
+    Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines.CSharpAddMissingInterpolationTokeFixer>;
+
+namespace Microsoft.CodeQuality.Analyzers.UnitTests.QualityGuidelines
+{
+    public class AddMissingInterpolationTokenTests
+    {
+        [Fact]
+        public async Task HasValidVariableInScope_Diagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine([|""{x}""|]);
+    }
+}";
+
+            var fixedCode = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine($""{x}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
+        }
+
+        [Fact]
+        public async Task DoesNotHaveValidVariableInScope_NoDiagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine(""{y}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, code);
+        }
+
+        [Fact]
+        public async Task ContainsOnlyLiterals_NoDiagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        Console.WriteLine(""{0}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, code);
+        }
+
+        [Fact]
+        public async Task ContainsLiteralAndBindableExpression_Diagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine([|""{x}, {0}""|]);
+    }
+}";
+
+            var fixedCode = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine($""{x}, {0}"");
```

Comment:
Not part of this PR/analyzer but I am wondering if we should have some analyzer to simplify some interpolated parts (like this `{0}`).

Reply:
@Evangelink Do you mean an analyzer for replacing `{0}` in interpolated string with just `0`? For this specific case, the behavior can differ per the current culture I believe.

Reply:
Yeah I meant that and you are probably right about the culture :)